meterfs/meterfs.c

```diff
@@ -447,29 +467,36 @@ int meterfs_readRecord(file_t *f, void *buff, size_t bufflen, unsigned int idx,
 		bufflen = f->header.recordsz - offset;
 
 	if (f->header.recordsz + sizeof(entry_t) <= sizeof(fbuff)) {
-		if ((stat = ctx->read(ctx->offset + addr, fbuff, f->header.recordsz + sizeof(entry_t))) < 0)
+		if ((stat = ctx->read(ctx->offset + addr, fbuff, f->header.recordsz + sizeof(entry_t))) < 0) {
+			meterfs_powerctrl(0, ctx);
 			return stat;
-		else
+		}
+		else {
 			read = stat - sizeof(entry_t);
-
+		}
 		if (eptr->id.nvalid || eptr->id.no != f->firstidx.no + idx)
 			return -ENOENT;
 
 		memcpy(buff, fbuff + sizeof(entry_t) + offset, bufflen);
 	}
 	else {
-		if ((stat = ctx->read(ctx->offset + addr + offsetof(entry_t, id), &id, sizeof(id))) < 0)
+		if ((stat = ctx->read(ctx->offset + addr + offsetof(entry_t, id), &id, sizeof(id))) < 0) {
+			meterfs_powerctrl(0, ctx);
 			return stat;
+		}
 		stat = 0;
 
 		if (id.nvalid || id.no != f->firstidx.no + idx)
 			return -ENOENT;
 
 		/* Read data */
-		if ((stat = ctx->read(ctx->offset + addr + sizeof(entry_t) + offset, buff, bufflen)) < 0)
+		if ((stat = ctx->read(ctx->offset + addr + sizeof(entry_t) + offset, buff, bufflen)) < 0) {
+			meterfs_powerctrl(0, ctx);
 			return stat;
-		else
+		}
+		else {
 			read = stat;
+		}
```

Comment:
Maybe just `read = ctx->read(ctx->offset + addr + sizeof(entry_t) + offset, buff, bufflen);`? We will return `read ` in just a moment anyway.

Reply:
done